amida-tech/indaba-client
# Task/inba 455 assessments

#### What's this PR do?
When a task is created, a call is made to the survey service, then greyscale backend. These calls create and store an assessmentId, associated to a task. However, if the user has not created a survey, it will prevent a task from being created, and display a toast pop up telling them to create a survey first. 

#### Related JIRA tickets:
This ticket has matching Greyscale changes at https://github.com/amida-tech/greyscale/pull/389

#### How should this be manually tested?

- Pull the changes for the Greyscale backend at https://github.com/amida-tech/greyscale/pull/389 
- Blow away the `indaba` and `indabatestuser` databases. (Note: If you're working on something critical with the backend right now, just change them to backups for the time being and rename them later.)
- Recreate `indaba` with the command `psql -h localhost -U indabauser indaba < schema.indaba.sql` 
- Recreate and populate `indabatestuser` with the command `mocha test\user.integration.js --bail -t 10000`. The backslash maybe different as you are on a Mac.
- Login as `test-adm@mail.net`.
- Create a new project, but skip creating a survey. 
- Create at least one stage and one subject. Create a couple of users, preferably with a userGroup.
- Go to the project in the Project Management page. 
- Drag and drop a name from the user panel onto the matrix. You should see a toast pop up and no task should be created.
- Click on the Survey name field and enter a name. 
- Drag and drop a name from the user field onto the matrix. A task should now appear.
- Go into your DB, and take a look at the `Tasks` table. Ensure that this newly created task has an `assessmentId` associated to it.

#### Any background context you want to provide?
#### Screenshots (if appropriate):
